src/main/java/net/codejava/MvcController.java

```diff
@@ -522,4 +542,149 @@ public String submitDispute(@ModelAttribute("user") User user) {
     return "account_info";
   }
 
+
+  /**
+   * HTML GET request handler that serves the "rewards_form" page to the user.
+   * An empty `User` object is also added to the Model as an Attribute to store
+   * the user's rewards form input.
+   * 
+   * @param model
+   * @return "rewards_form" page
+   */
+  @GetMapping("/rewards")
+	public String showRewardsForm(Model model) {
+    User user = new User();
+		model.addAttribute("user", user);
+		return "rewards_form";
+	}
+
+  /**
+   * HTML POST request handler for the Rewards Form page.
+   * 
+   * The same username+password handling from the login page is used.
+   * 
+   * If the password attempt is correct, the reward action is taken.
+   * The user is then served the "account_info" with an updated balance.
+   * 
+   * If the password attempt is incorrect, the user is redirected to the "welcome" page.
+   * 
+   * @param user
+   * @return "account_info" page if login successful. Otherwise, redirect to "welcome" page.
+   */
+  @PostMapping("/rewards")
+  public String submitRewards(@ModelAttribute("user") User user) {
+    String userID = user.getUsername();
+    String userPasswordAttempt = user.getPassword();
+
+    String getUserPasswordSql = String.format("SELECT Password FROM passwords WHERE CustomerID='%s';", userID);
+    String userPassword = jdbcTemplate.queryForObject(getUserPasswordSql, String.class);
+
+    // unsuccessful login
+    if (userPasswordAttempt.equals(userPassword) == false) {
+      return "welcome";
+    }
+
+    String userRewardChoice = user.getRewardChoice();
+    int userPointsToExchange = user.getPointsToExchange();
+    
+    String currentTime = SQL_DATETIME_FORMATTER.format(new java.util.Date());
+
+    String getUserRewardPointsSql = String.format("SELECT RewardPoints FROM customers WHERE CustomerID='%s';", userID);
+    int userRewardPoints = jdbcTemplate.queryForObject(getUserRewardPointsSql, Integer.class);
+
+    // Inputs more points than owned
+    if (userPointsToExchange > userRewardPoints) {
+      return "welcome";
+    }
+
+    // if the overdraft balance is positive, subtract the deposit with interest
+    if (userRewardChoice.equals("Fraud Reversal") && userRewardPoints >= 500) {

```

Comment:
Make a constant for 500 and use that instead

Reply:
Done